compiler/test/compilable/test23978.d

```diff
@@ -0,0 +1,29 @@
+// REQUIRED_ARGS: -preview=dip1021 -lowmem
+// https://issues.dlang.org/show_bug.cgi?id=23978
+
+// Note: the bug involved memory corruption because a `malloc`ed `EscapeBy[]` array
+// wasn't scanned by GC, resulting in a segfault. If this fails again, it might not
+// be consistent, so try running it a few times.
```

Comment:
Not sure I agree with this rationale. Where is the C malloc'd memory in EscapeBy?

Reply:
I've updated the comment